### PR TITLE
Move FC-quickstarts and update Basic assembly

### PR DESCRIPTION
**Pull request by hamishwillee**
This moves the quickstarts under their associated flight controller out from the "Basic Assembly" section, and then updated the guide to link to them and also to the build logs. In other words this is now a single page that is all about assembly.

I would like to create a quickstart that is generic, but before I do that my intent is to work out how this Assembly doc can split into the different vehicle sections with single source.
Once that's done we are on final stage to pull MC, VTOL, Plane to the top level.

- [ ] Before this merges need to check all links to `assembly/index.md` make sense with this change.
- [ ] Work out _single source_ options.

### CODE REVIEW

**Comment by sfuhrer** [comment]
Good stuff, agree with the direction. Please let me know once you need proper reviews.

**Comment by hamishwillee** [comment]
Thanks @sfuhrer 

I'm not quite ready for reviews, but I am ready for advice.

I'm at the point of talking about actuators and I am wondering what to recommend for the different vehicle types - because this is a bit vehicle specific.

My thinking is that an MC outputs is all about motors. Generally you might have a power distribution board  with a BEC for powering winches/other peripherals (which you'd connect to the power rail),
- Can I assume that most peripherals all use the same voltage? Is the BEC voltage usually configurable on a PDB or ESC?
- I'd like to talk about PDB here because its an easy recommendation, but is it better to talk about a more general case such as how you wire this up yourself?

What about VTOL - do they use PDBs. My assumption is probably yes - at least up to 4 motors.

Just trying to get a sense of what we should recommend

**Comment by hamishwillee** [comment]
@sfuhrer Can you please review the _assembly.md doc. This is what will be duplicated for the different vehicles. So this one https://github.com/PX4/PX4-user_guide/blob/42155ad8de344c5c75046dd63a272101545aeb14/en/assembly/_assembly.md

I've still got work to do "around" this, but good to check the core content. 

Note, it isn't very specific to a vehicle type, but it does have some customisation.

**Review by sfuhrer — APPROVED** [review]
I like it, it's very easy to read but hopefully gives users a good starting point. Let's see how much we can to customize it for MC vs FW vs VTOL.

File: en/assembly/_assembly.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+
+<img src="../../assets/assembly/pixhawk6x_holybro.jpg" width="300px" title="Holybro Pixhawk6x" /><img src="../../assets/assembly/pixhawk6x_cuav.jpg" width="300px" title="Cuav Pixhawk6x" />
+
+The connector standard ports are listed below, along with their labels on each of the FCs, and what they are used for.
+The FCs have much the same ports with similar names, and core peripherals are connected in the same way.
+
+| Standard                    | Holybro       | CUAV                | Connect to ...                                               |
+| --------------------------- | ------------- | ------------------- | ------------------------------------------------------------ |
+| Full GPS plus Safety Switch | GPS1          | GPS&SAFETY          | Primary GPS (gps, compass, safety switch, buzzer, LED)       |
+| Basic GPS                   | GPS2          | GPS2                | Secondary GPS (GPS/compass)                                  |
```
Comment (hamishwillee) [inline en/assembly/_assembly.md:28]:
```suggestion
| Basic GPS                   | GPS2          | GPS2                | Secondary GNSS module (GNSS/compass)                                  |
```

File: en/assembly/_assembly.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+[RTK GNSS](../gps_compass/rtk_gps.md) modules with centimetre accuracy are also supported.
+
+GNSS/Compass modules [should be mounted on the frame](../assembly/mount_gps_compass.md) as far away from other electronics as possible, with the direction marker pointing towards the front of the vehicle (separating the compass from other electronics will reduce electromagnetic interference).
+
+The modules connect to either serial ports or to the CAN bus.
+
+### Serial Port GNSS Modules
+
+The primary GNSS/compass module is connected to the flight controller ort labeled `GPS1` or `GPS&SAFETY`, which should be a 10-Pin JST GH connector.
+A second GPS/compass module, if present, is attached to the 6-pin port labeled `GPS2`.
```
Comment (hamishwillee) [inline en/assembly/_assembly.md:69]:
```suggestion
A second GNSS/compass module, if present, is attached to the 6-pin port labeled `GPS2`.
```

File: en/assembly/_assembly.md [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+Their available ports are very similar because they are both [Pixhawk standard autopilots](../flight_controller/autopilot_pixhawk_standard.md) that have adopted the [Pixhawk connector standard](https://github.com/pixhawk/Pixhawk-Standards/blob/master/DS-009%20Pixhawk%20Connector%20Standard.pdf).
+
+<img src="../../assets/assembly/pixhawk6x_holybro.jpg" width="300px" title="Holybro Pixhawk6x" /><img src="../../assets/assembly/pixhawk6x_cuav.jpg" width="300px" title="Cuav Pixhawk6x" />
+
+The connector standard ports are listed below, along with their labels on each of the FCs, and what they are used for.
+The FCs have much the same ports with similar names, and core peripherals are connected in the same way.
+
+| Standard                    | Holybro       | CUAV                | Connect to ...                                               |
+| --------------------------- | ------------- | ------------------- | ------------------------------------------------------------ |
+| Full GPS plus Safety Switch | GPS1          | GPS&SAFETY          | Primary GPS (gps, compass, safety switch, buzzer, LED)       |
```
[the remark this answers is not in the document]
Comment (hamishwillee) [reply en/assembly/_assembly.md:27]: Yes, we should use GNSS everywhere we can. There are plenty of places we can't, such as labels on FCs, and product names.

File: en/assembly/_assembly.md [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+  - [Mounting a Compass](../assembly/mount_gps_compass.md)
+
+<div v-if="$frontmatter.frame === 'Multicopter'">
+
+::: info
+The and sections of [PX4 Basic Concepts](../getting_started/px4_basic_concepts.md) provide a broad overview of the main drone components.
+[Hardware Selection & Setup](../hardware/drone_parts.md) contains details about specific hardware, including selection, assembly, and configuration.
+:::
+
+- [Multicopter Racer Setup](../config_mc/racer_setup.md) — racer-specific assembly and configuration information (racers don't have GPS module)
```
Comment (hamishwillee) [inline en/assembly/_assembly.md:311]:
```suggestion
- [Multicopter Racer Setup](../config_mc/racer_setup.md) — racer-specific assembly and configuration information (racers don't usually have a GNSS module)
```

File: en/assembly/index.md [hunk 5]
```diff
@@ -1,14 +1,80 @@
...
 
-This section contains topics that explain how to assemble such a system for different [flight controllers](../flight_controller/index.md).
+::: info
+The [Drone Components & Parts](../getting_started/px4_basic_concepts.md#drone-components-parts) and [Payloads](../getting_started/px4_basic_concepts.md#payloads) sections of [PX4 Basic Concepts](../getting_started/px4_basic_concepts.md) provide a broad overview of the main drone components.
+[Hardware Selection & Setup](../hardware/drone_parts.md) contains details about specific hardware, including selection, assembly, and configuration.
+:::
+
+## Overview
+
+The core parts of a _typical_ PX4-based autopilot system consist of a flight controller, GPS, external compass, manual controller and/or telemetry radio system, motors and/or other control actuators, payloads, and a power system.
```
[the remark this answers is not in the document]
Comment (hamishwillee) [reply en/assembly/index.md:12]: I'm actually hoping to kill this doc sorry - the entry point would be the other doc. This was a placeholder/first effort.

[the remark this answers is not in the document]
Comment (hamishwillee) [reply en/assembly/index.md:12]: But yes, we should. I'll ask questions on this in slack.

File: en/assembly/index.md [hunk 6]
```diff
@@ -1,14 +1,80 @@
...
+A VTOL or Fixed-wing vehicle will generally also have an airspeed sensor.
+
+The process for assembling the components together "off the frame" is similar most flight controllers, and in particular for those that follow the Pixhawk connector standard.
+The standard specifies flight controller connectors for attaching the GPS module, external compass, telemetry radio, analog power system, and so on, and many component vendors in the ecosystem provide compatible cables.
```
Comment (hamishwillee) [inline en/assembly/index.md:16]:
```suggestion
The standard specifies flight controller connectors for attaching the GNSS module, external compass, telemetry radio, analog power system, and so on, and many component vendors in the ecosystem provide compatible cables.
```

File: en/assembly/index.md [hunk 7]
```diff
@@ -1,14 +1,80 @@
...
+- [Tailsitter VTOL](../frames_vtol/tailsitter.md)
+- [Tiltrotor VTOL](../frames_vtol/tiltrotor.md)
+
+Other frames:
+
+- [Airframe Builds](../airframes/index.md) for complete assembly examples on different vehicle frames.
+
+## See Also
+
+- [Multicopter Racer Setup](../config_mc/racer_setup.md) — racer-specific assembly and configuration information (racers don't have GPS module)
```
Comment (hamishwillee) [inline en/assembly/index.md:79]:
```suggestion
- [Multicopter Racer Setup](../config_mc/racer_setup.md) — racer-specific assembly and configuration information (racers don't usually have a GNSS module)
```

File: en/assembly/_assembly.md [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+::: info
+If the controller cannot be mounted in the recommended/default orientation (e.g. due to space constraints) you will need to configure the autopilot software with the orientation that you actually used: [Flight Controller Orientation](../config/flight_controller_orientation.md).
+:::
+
+## GNSS/Compass (+ Buzzer + Safety Switch + LED)
+
+Pixhawk Autopilot systems usually include a primary [GNSS/Compass module](../gps_compass/#supported-gnss) that incorporates a [buzzer](../getting_started/px4_basic_concepts.md#buzzer), [safety switch](../getting_started/px4_basic_concepts.md#safety-switch), and [UI LED](../getting_started/led_meanings.md#ui-led), and may also have a secondary GPS/compass module.
+[RTK GNSS](../gps_compass/rtk_gps.md) modules with centimetre accuracy are also supported.
+
+GNSS/Compass modules [should be mounted on the frame](../assembly/mount_gps_compass.md) as far away from other electronics as possible, with the direction marker pointing towards the front of the vehicle (separating the compass from other electronics will reduce electromagnetic interference).
```
[the remark this answers is not in the document]
Comment (hamishwillee) [reply en/assembly/_assembly.md:62]: I know, but this is the easiest way of mounting so you don't have to explain the detail, and also for users so they don't have to configure anything
I'll tweak.

```suggestion
GNSS/Compass modules should be mounted on the frame as far away from other electronics as possible — separating the compass from other electronics will reduce electromagnetic interference.
Provided the direction marker is oriented upright and facing towards the front of the vehicle, or at some 45 degrees multiple offset in any axis, the orientation will be auto-detected during calibration.
If needed you can orient at other offsets, but then you'll need to manually set the orientation.
For more information see [Mounting a Compass (or GNSS/Compass)](../assembly/mount_gps_compass.md).
```

File: .vitepress/config.mjs [hunk 9]
```diff
@@ -43,8 +43,16 @@ export default defineConfig({
       // other locale specific properties...
       themeConfig: {
         sidebar: getSidebar.sidebar("en"),
+
         editLink: {
-          pattern: "https://github.com/PX4/PX4-user_guide/edit/main/:path",
+          pattern: ({ filePath, frontmatter }) => {
```
Comment (hamishwillee) [inline .vitepress/config.mjs:48]: This allows us to specify `newEditLink` in page metadata to fix up the edit link for a page that includes most of its data from another page.

**Comment by hamishwillee** [comment]
Thanks for review @sfuhrer 

FYI, I have done more work on this today, starting the process of pulling out the MC, VTOL, FW to the top level. I might merge tomorrow. It does highlight that we need a more coherent config guide for VTOL/FW, somewhat like what we "are getting there" for MC.

I think this will be easier with the vehicle sections published - i.e. it will force us to focus on this stuff :-). 

Bit more to do with the introduction and operations too to make this all work.

**Pull request merged**